Write security considerations, cipher suites registry (#42)
ietf-wg-keytrans/draft-protocol: draft-ietf-keytrans-protocol.md

    applicability may be recommended, but in such cases applicability statements
    that describe any limitations of the mechanism or necessary constraints will
    be provided.
  - N: Indicates that the item's associated mechanism has not been evaluated and
    is not RECOMMENDED (as opposed to being NOT RECOMMENDED). This does not
    mean that the mechanism is flawed. For example, an item may be marked as "N"
    because it has usage constraints or limited applicability.
  - D: Indicates that the item is discouraged and SHOULD NOT be
    used. This marking could be used to identify mechanisms that might result in
    problems if they are used, such as a weak cryptographic algorithm or a
    mechanism that might cause interoperability problems in deployment.
- Reference: The document where this cipher suite is defined.

Initial contents:

| Value  | Name                                 | R | Ref      |
|:-------|:-------------------------------------|:--|:---------|
| 0x0000 | RESERVED                             | - | RFC XXXX |
| 0x0001 | KT_128_SHA256_P256                   | Y | RFC XXXX |
| 0x0002 | KT_128_SHA256_Ed25519                | Y | RFC XXXX |

> For all cipher suites defined in this document specifically, the following
> primitives and parameters are the same:

Reviewer: ```suggestion
All cipher suites currently specified share the following primitives and
parameters:
```

- The hash algorithm is SHA-256, as defined in {{SHS}}.
- `Nc`: 16

> - `Kc`: The following string, hex decoded: `d821f8790d97709796b4d7903357c3f5`

Reviewer: ```suggestion
- `Kc`: The byte sequence equal to the hex-encoded string `d821f8790d97709796b4d7903357c3f5`
```

## KT Cipher Suites

A cipher suite is a specific combination of cryptographic primitives and
parameters to be used in an instantiation of the protocol. Cipher suite names
follow the naming convention:

~~~ tls-presentation
CipherSuite KT_LVL_HASH_SIG = VALUE;
~~~

Where `VALUE` is represented as a 16-bit integer:

~~~ tls-presentation
uint16 CipherSuite;
~~~

The columns in the registry are as follows:

- Value: The numeric value of the cipher suite.
- Name: The name of the cipher suite.
- Recommended: Whether support for this cipher suite is recommended by the IETF.
  Valid values are "Y", "N", and "D", as described below. The default value of
  the "Recommended" column is "N". Setting the Recommended item to "Y" or "D",
  or changing an item whose current value is "Y" or "D", requires Standards
  Action {{!RFC8126}}.
  - Y: Indicates that the IETF has consensus that the item is RECOMMENDED. This
    only means that the associated mechanism is fit for the purpose for which it

>     was defined. Careful reading of the documentation for the mechanism is

Author: The applicability / purpose of a mechanism is specified in an architecture document. So like, the arch document we have for keytrans

  - N: Indicates that the item has not been evaluated by the IETF and that the
    IETF has made no statement about the suitability of the associated
    mechanism. This does not necessarily mean that the mechanism is flawed, only
    that no consensus exists. The IETF might have consensus to leave an item
    marked as "N" on the basis of it having limited applicability or usage
    constraints.
  - D: Indicates that the item is discouraged and SHOULD NOT or MUST NOT be
    used. This marking could be used to identify mechanisms that might result in
    problems if they are used, such as a weak cryptographic algorithm or a
    mechanism that might cause interoperability problems in deployment.
- Reference: The document where this cipher suite is defined.

Initial contents:

| Value  | Name                                 | R | Ref      |
|:-------|:-------------------------------------|:--|:---------|
| 0x0000 | RESERVED                             | - | RFC XXXX |
| 0x0001 | KT_128_SHA256_P256                   | Y | RFC XXXX |
| 0x0002 | KT_128_SHA256_Ed25519                | Y | RFC XXXX |

For all cipher suites this document defines, the following primitives and

> parameters are the same:

Author: This statement only holds for this document. Other documents may create cipher suites that do something different

- The hash algorithm is SHA-256, as defined in {{SHS}}.
- `Nc`: 16

> - `Kc`: The hex-decoded value: `d821f8790d97709796b4d7903357c3f5`

Author: We are not standardizing that all cipher suites must use SHA-256? We are just saying that the ones defined here do.

The signature algorithm and VRF algorithm for each cipher suite is as follows:

| Name                  | Signature              | VRF Algorithm                     |
|:----------------------|:-----------------------|:----------------------------------|
| KT_128_SHA256_P256    | ecdsa_secp256r1_sha256 | ECVRF-P256-SHA256-TAI             |
| KT_128_SHA256_Ed25519 | ed25519                | ECVRF-EDWARDS25519-SHA512-TAI[32] |

The VRF algorithms are specified in {{RFC9381}}. For `KT_128_SHA256_Ed25519`,
the final hash output of `ECVRF-EDWARDS25519-SHA512-TAI` is truncated to be 32
bytes.

> ## KT Designated Expert Pool {#de}

Reviewer: Oy. This seems like quite the process being standardized here. Has this already been worked towards? This must be carefully discussed before we proceed. I would like to learn more about the requirements for having a "designated experts" group etc. Why not just leave it up to the WG?

Author: I'm pretty sure it is just a mailing list to approve allocating codepoints. TLS, MLS, etc have them. The keytrans wg may not always exist to make such decisions
